File: frontend/src/modules/graph/application/use-cases/discardAll.ts
import { WorkingChangesWriter } from "@modules/graph/application/contracts/WorkingChangesWriter";

export async function discardAll(
  workingChangesWriter: WorkingChangesWriter,
  repositoryPath: string
): Promise<void> {
  return workingChangesWriter.discardAll(repositoryPath);
}

File: backend/src/app/working_changes/discard_all.rs
use std::path::Path;

use crate::app::working_changes::contracts::WorkingChangesWriter;
use crate::domain::errors::AppError;

pub fn discard_all(
    working_changes_writer: &dyn WorkingChangesWriter,
    repository_path: &Path,
) -> Result<(), AppError> {
    working_changes_writer.discard_all(repository_path)
}
